src/simulations/collect.jl

```diff
@@ -1,12 +1,20 @@
 export run!, collect_agent_data!, collect_model_data!,
        init_agent_dataframe, init_model_dataframe, aggname,
-       should_we_collect
+       should_we_collect, Identity, Copy, DeepCopy
+
+abstract type Obtainer end
+struct Identity <: Obtainer end
+struct Copy <: Obtainer end
+struct DeepCopy <: Obtainer end
```

Comment:
Tim, why go through make *new* names, specifically, introduce new types? The user can directly give `copy` or `deepcopy` to a keyword `obtainer`, and we can just use this value to the functions. E.g. we have
```julia
get_data(a, f::Function, obtainer = identity) = obtainer(f(a))
get_data(a, s::Symbol, obtainer = identity) = obtainer(getproperty(a, s))
```

Reply:
and `obtainer = identity` becomes a keyword of `run!`.

Reply:
Oh. That's what you meant by identity! `identity()`, not just a keyword named identity. 

Yep. Sounds far better. I'll revert a bunch of this.

Reply:
haha, sorry, I should have been clearer!